include/dlaf/matrix/panel.h

```diff
@@ -97,28 +98,53 @@ struct Panel<axis, const T, device> : protected Matrix<T, device> {
     }
     else {
       internal_.insert(internal_linear_idx);
-      return BaseT::read(fullIndex(index));
+      return data_.read(fullIndex(index));
     }
   }
 
+  void setRange(LocalTileSize start, LocalTileSize end) noexcept {
+    const auto start_loc = start.get(component(axis));
+    const auto end_loc = end.get(component(axis));
+
+    DLAF_ASSERT(start_loc < end_loc, start_loc, end_loc);
+
+    DLAF_ASSERT(start_loc >= bias_, start, bias_);
+    DLAF_ASSERT(end_loc <= dist_matrix_.localNrTiles().get(component(axis)), end,
+                dist_matrix_.localNrTiles().get(component(axis)));
+
+    start_ = start_loc;
+    end_ = end_loc;
+  }
+
```

Comment:
Maybe just call setRangeStart and setRangeEnd...

Reply:
I started with the solution you are proposing, but the assertions are a bit different.

By changing separately `start` or `end`, they have to be coherent with the other which is already set, while changing all together in a single step, they have to respect the "global" constraint, in addition to be coherent as a pair (but not considering existing setup).

Hope it is clear enough why I opted for the "duplicated" code, but I'm open to any idea that can improve this.